frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -290,6 +290,11 @@ const TextField = ({
             {...props}
           />
         )}
+        onChange={(_e, v) => {
+          if (v && onReturn) {
+            onReturn();
```

Comment:
Wonder if we should modify onReturn to send back the value, might allow us to modify catalog/selector to handle the case at that point.

When you said it doesn't work for catalog/selector did that mean it breaks them or they still retain the same functionality and just not the extended functionality?

Reply:
Yeah I'm curious why this only works for some of the use cases

Reply:
it means they still retain the same functionality, just not the extended functionality

Reply:
@dschaller the reason why those are different is because they set the `value` prop. The resolver and k8s dashboard do not set this prop which is why they work. They also use `formRegistration` but I don't think that matters. All the other props are the same between the 4 cases (resolver and k8s dashboard work, proj catalog and proj selector do not work).

Reply:
Ok I think this is fine for a partial fix but can we add a TODO to come back and investigate the other use cases